### PR TITLE
Ra/test min api

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Moves #23907 to this branch so I can push commits.

### CODE REVIEW

File: aspnetcore/test/integration-tests.md [hunk 1]
```diff
@@ -118,6 +118,44 @@ If the SUT's [environment](xref:fundamentals/environments) isn't set, the enviro
...
+ASP.NET Core 6 introduced [`WebApplication`](/dotnet/api/microsoft.aspnetcore.builder.webapplication) which removed the need for a `Startup` class. To test with `WebApplicationFactory` without a `Startup` class, an ASP.NET Core 6 app needs to expose the implicitly defined `Program` class to the test project in **one** of the following ways:
+
+* Expose internal types from the web app to the test project. This can be done in the project file (`.csproj`):
+  ```xml
+  <ItemGroup>
+       <InternalsVisibleTo Include="MyTestProject" />
+  </ItemGroup>
+  ```
+* Make the `Program` class public using a partial class declaration
+:
```
Comment (halter73) [inline aspnetcore/test/integration-tests.md:132]:
```suggestion
* Make the `Program` class public using a partial class declaration:
```

File: aspnetcore/test/integration-tests.md [hunk 2]
```diff
@@ -118,6 +118,44 @@ If the SUT's [environment](xref:fundamentals/environments) isn't set, the enviro
...
+  ```diff
+  var builder = WebApplication.CreateBuilder(args);
+  // ... Configure services, routes, etc.
+  app.Run();
+  + public partial class Program { }
+  ```
+
+After making the changes in the Web application. The test project now can use the `Program` class for the `WebApplicationFactory`.
```
Comment (halter73) [inline aspnetcore/test/integration-tests.md:140]:
```suggestion
After making the changes in the web application, the test project now can use the `Program` class for the `WebApplicationFactory`.
```

File: aspnetcore/test/integration-tests.md [hunk 3]
```diff
@@ -118,6 +118,44 @@ If the SUT's [environment](xref:fundamentals/environments) isn't set, the enviro
...
+{
+    var application = new WebApplicationFactory<Program>()
+        .WithWebHostBuilder(builder =>
+        {
+            // ... Configure test services
+        });
+        
+    var client = application.CreateClient();
+    ...
+    }
```
Comment (halter73) [inline aspnetcore/test/integration-tests.md:154]:
```suggestion
    //...
}
```

**Comment by Rick-Anderson** [comment]
@furoTmark 
Thanks for the fix. Be sure to check out your name as a contributor about the 3rd of next month on the [What's new page](https://docs.microsoft.com/en-us/aspnet/core/whats-new/)

**Comment by furoTmark** [comment]
Didn't see my name, but glad to help!

**Pull request merged**